core/src/main/scala/org/http4s/otel4s/middleware/HttpAttributes.scala

```diff
@@ -16,206 +16,204 @@
-      Attribute("http.request_content_length", cl)
-    def responseContentLength(cl: Long): Attribute[Long] =
-      Attribute("http.response_content_length", cl)
-    def retryCount(i: Int): Attribute[Long] = Attribute("http.retry_count", i.toLong)
-    def peerIp(ip: IpAddress): Attribute[String] =
-      Attribute("net.peer.ip", ip.toString()) // TODO: Check that this is the right way
-    def peerPort(port: Port): Attribute[Long] = Attribute("net.peer.port", port.value.toLong)
-  }
+object HttpAttributes {
+  def httpRequestMethod(method: Method): Attribute[String] =
+    SemanticAttributes.HttpRequestMethod(method.name)
+  def httpRequestResendCount(count: Long): Attribute[Long] =
+    SemanticAttributes.HttpRequestResendCount(count)
+  def httpResponseStatusCode(status: Status): Attribute[Long] =
+    SemanticAttributes.HttpResponseStatusCode(status.code.toLong)
+  def networkPeerAddress(ip: IpAddress): Attribute[String] =
+    SemanticAttributes.NetworkPeerAddress(ip.toUriString)
+  def serverAddress(host: Host): Attribute[String] =
+    SemanticAttributes.ServerAddress(Host.headerInstance.value(host))
+  def urlFull(url: Uri): Attribute[String] =
+    SemanticAttributes.UrlFull(url.renderString)
+  def urlPath(path: Uri.Path): Attribute[String] =
+    SemanticAttributes.UrlPath(path.renderString)
+  def urlQuery(query: Query): Attribute[String] =
+    SemanticAttributes.UrlQuery(query.renderString)
+  def urlScheme(scheme: Uri.Scheme): Attribute[String] =
+    SemanticAttributes.UrlScheme(scheme.value)
+  def userAgentOriginal(userAgent: `User-Agent`): Attribute[String] =
+    SemanticAttributes.UserAgentOriginal(`User-Agent`.headerInstance.value(userAgent))
 
   object Client {
-    def peerName(uri: Uri): Option[Attribute[String]] =
-      uri.host.map(h => Attribute("net.peer.name", h.value))
+    def serverPort(port: Port): Attribute[Long] =
+      SemanticAttributes.ServerPort(port.value.toLong)
   }
 
   object Server {
-    def serverName(s: String): Attribute[String] = Attribute("http.server_name", s)
-    // The route template. Since http4s uses unapplies by default this is non-trivial.
-    // Accept-List for segements are better to get coherent routes, but block-list segments
-    // take much less effort while allowing unexpected variables. We will provide options
-    // for this
-    def route(s: String): Attribute[String] = Attribute("http.route", s)
-    // This is either the net ip, OR the x-forwarded for depending on what is available.
-    def clientIp(ip: IpAddress): Attribute[String] = Attribute("http.client_ip", ip.toString())
+    def clientAddress(
+        forwarded: Option[`X-Forwarded-For`],
+        ip: Option[IpAddress],
+    ): Option[Attribute[String]] = {
+      @inline def fromIp(ip: IpAddress): Attribute[String] =
+        SemanticAttributes.ClientAddress(ip.toUriString)
+      forwarded
+        .map(_.values.toList.flatten.headOption.map(fromIp))
+        .getOrElse(ip.map(fromIp))
+    }
```

Comment:
this one is is definitely difficult and worth having a second look (see also the docs on `SemanticAttributes.ClientAddress`) @rossabaker

Reply:
logic changed to only use the first IP from `X-Forwarded-For`